eng/scripts/verify-npm-tags.ps1

```diff
@@ -0,0 +1,46 @@
+param (
+  [Parameter(mandatory = $true)]
+  $originalDistTags,
+  [Parameter(mandatory = $true)]
+  $intendedTag,
+  [Parameter(mandatory = $true)]
+  $intendedTagVersion,
+  [Parameter(mandatory = $true)]
+  $packageName,
+  [Parameter(mandatory = $true)]
+  $npmToken
+)
+
+$ErrorActionPreference = 'Stop'
+$PSNativeCommandUseErrorActionPreference = $true
+
+Write-Host "Verify npm tag versions for package $packageName"
+
+$parsedOriginalDistTags = $originalDistTags | ConvertFrom-Json
+
+$npmPkgProp = npm view $packageName --json | ConvertFrom-Json
+$packageDistTags = $npmPkgProp."dist-tags"
+
+Write-Host "Original dist-tag: $parsedOriginalDistTags"
+Write-Host "Current dist-tag: $packageDistTags"
+Write-Host "Intend to add tag $intendedTag to version $intendedTagVersion"
+
+if ($packageDistTags."$intendedTag" -ne $intendedTagVersion) {
+  Write-Warning "Tag not correctly set, current $intendedTag tag is version $($packageDistTags."$intendedTag") instead of $intendedTagVersion."
+  $correctDistTags = $parsedOriginalDistTags
+  $correctDistTags."$intendedTag" = $intendedTagVersion
+
+  Write-Host "Setting AuthToken Deployment"
+  $regAuth = "//registry.npmjs.org/"
+  npm config set $regAuth`:_authToken=$npmToken
```

Comment:
As a follow-up question. Does this npm config file write something to disk? As in does it make an npmrc file with this authToken in it? If so perhaps that is a reason to keep it an env variable so that secret is only in the env and not on the file system.

Reply:
yes it does mutate the .npmrc file